### PR TITLE
fix: update available properties

### LINKED ISSUE #2123: Check that Configuration Properties page documents all properties
From a comment from Simon Martinelli in https://vaadin.com/docs/latest/configuration/properties

> Some default values are missing. For example what's the default value for heartbeatInterval? And it's unclear if the interval is milliseconds, seconds or minutes.

### PR DESCRIPTION
**Pull request by caalador**
Update properties that are available
and add default value for each property.

Fixes #2123

### CODE REVIEW

File: articles/configuration/properties.asciidoc [hunk 1]
```diff
@@ -102,70 +102,99 @@ The following list contains the properties that are defined in the [classname]`c
...
+|`requestTiming`
+|If this is set to `true`, the server includes some basic timing information in each response. This can be used for performance testing.
+|`true` for development mode, `false` for production mode
+
+|`sendUrlsAsParameters`
+|Returns `true` if the sending of URLs as GET and POST parameters in requests with content-type `application/x-www-form-urlencoded` is enabled.
+|`true`
+
+|`syncIdCheck`
+|Returns whether sync ID checking is enabled. The sync ID is used to handle situations in which the client sends a message to a connector that has been removed recently from the server.
```
Comment (mcollovati) [inline articles/configuration/properties.asciidoc:188]:
```suggestion
|Enables or disables the sync ID checking. The sync ID is used to handle situations in which the client sends a message to a connector that has been removed recently from the server.
```

File: articles/configuration/properties.asciidoc [hunk 2]
```diff
@@ -102,70 +102,99 @@ The following list contains the properties that are defined in the [classname]`c
...
+|`true`
+
+|`useDeprecatedV14Bootstrapping`
+|This flag can be used to enable the server-side bootstrapping mode, which was used in Vaadin 14 and earlier versions. This option is only supported if webpack is used as the front-end build tool and not if the application uses Vite, which is the default. You can <<{articles}/configuration/live-reload#webpack-feature-flag,enable webpack using its associated feature flag>>.
+|`false` (mode removed in v24)
+
+|`webComponentDisconnect`
+|Returns the number of seconds that a WebComponent will wait for a reconnect before removing the server-side component from memory.
```
Comment (mcollovati) [inline articles/configuration/properties.asciidoc:196]:
```suggestion
|Defines the number of seconds that a WebComponent will wait for a reconnect before removing the server-side component from memory.
```

File: articles/configuration/properties.asciidoc [hunk 3]
```diff
@@ -102,70 +102,99 @@ The following list contains the properties that are defined in the [classname]`c
...
+|Property name
+|Description
+|Default value
+
+|`brotli`
+|Checks whether pre-compressed Brotli files should be used if available.
+|`true`
+
+|`closeIdleSessions`
+|When set to `true`, the session is closed if no UI is active. From the servlet container's viewpoint, heartbeat requests are like any other request. This means that as long as there is an open UI, the session never expires, even if there is no user interaction. You can control this behavior by setting the init parameter named `closeIdleSessions` to `true`.
```
[the remark this answers is not in the document]
Comment (caalador) [reply articles/configuration/properties.asciidoc]: Yes. Also the description is a hard to follow so I updated it a bit.

File: articles/configuration/properties.asciidoc [hunk 4]
```diff
@@ -102,70 +102,99 @@ The following list contains the properties that are defined in the [classname]`c
...
+|`devmode.liveReload.enabled`
+|This means that if you are using some live reload tool on the server side, the browser is refreshed automatically after the code is reloaded on the server side.
+|`true`
+
+|`devmode.optimizeBundle`
+|By default, in development mode all front-end resources found on the class path are included in the generated webpack bundle. When set to `true`, this creates an optimized bundle by including only front-end resources that are used from the application entry points. It uses bytecode scanning, which increases application start-up time.
+|`false` for development mode, `true` for production mode
+
+|`devmode.sessionSerialization.enabled`
+|This determines whether the [classname]`VaadinSession` instances of the users are serialized on server shutdown when running in development mode. When the parameter is set to `false`, only other HTTP session data is serialized or deserialized on development server restart. Enabling the property allows, for example, access control information to be preserved during development, so that you don't need to log in again for each change.
```
[the remark this answers is not in the document]
Comment (caalador) [reply articles/configuration/properties.asciidoc]: Updated description

File: articles/configuration/properties.asciidoc [hunk 5]
```diff
@@ -102,70 +102,99 @@ The following list contains the properties that are defined in the [classname]`c
...
+|`productionMode`
+|This sets the application to work in production mode. Production mode disables most of the logged information that appears on the console. Logging and other debugging features can have a significant impact on performance otherwise. Development-mode JavaScript functions aren't exported. A `push` is given as a minified JavaScript file instead of a full size one, and static resources are cached. See <<../production#,Deploying to Production>> for more information.
+|`false`
+
+|`pushLongPollingSuspendTimeout`
+|When using the long polling transport strategy, this specifies how long it accepts responses after each network request, in milliseconds.
+|`-1` (no timeout)
+
+|`pushMode`
+|The permitted values are "disabled" or "manual". See <<{articles}/advanced/server-push#, Server Push>> for more information.
```
[the remark this answers is not in the document]
Comment (caalador) [reply articles/configuration/properties.asciidoc]: Added.

File: articles/configuration/properties.asciidoc [hunk 6]
```diff
@@ -102,70 +102,99 @@ The following list contains the properties that are defined in the [classname]`c
...
+|`pushLongPollingSuspendTimeout`
+|When using the long polling transport strategy, this specifies how long it accepts responses after each network request, in milliseconds.
+|`-1` (no timeout)
+
+|`pushMode`
+|The permitted values are "disabled", "manual" or "automatic". See <<{articles}/advanced/server-push#, Server Push>> for more information.
+|`disabled`
+
+|`pushServletMapping`
+|Returns the servlet mapping that bidirectional ("push") client-server communication should use.
```
Comment (mcollovati) [inline articles/configuration/properties.asciidoc:176]: Sorry, missed that one on the previous review
```suggestion
|Sets the servlet mapping that bidirectional ("push") client-server communication should use.
```

**Review by mcollovati — CHANGES_REQUESTED** [review]

**Review by mcollovati — APPROVED** [review]
